src/mbgl/util/i18n.cpp

```diff
@@ -627,4 +628,17 @@ bool isWhitespace(char16_t chr) {
 
 } // namespace i18n
 } // namespace util
+
+GlyphIDType charGlyphIDType(char16_t ch, GlyphIDType lastChType) {
+    if (u'\uFE00' == ch) {
+        return lastChType;
+    }
+
+    if (isInKhmer(ch)) return GlyphIDType::Khmer;
+    if (isInMyanmar(ch) || isInMyanmarExtendedA(ch) || isInMyanmarExtendedB(ch)) return GlyphIDType::Myanmar;
+    if (isInDevanagari(ch) || isInDevanagariExtended(ch)) return GlyphIDType::Devanagari;
```

Comment:
Here the mapping from unicode range to font file is hard coded. For every new language we would need to add an entry.

Would it make more sense to specify in the font definition what unicode range it covers:

So instead of 

```json
"fonts": "https://alanchenboy.github.io/harfbuzzResource/fonts/{fontstack}/{language}.ttf"
```

we could do this:

```json
"fonts": [
  {
    "name": "devanagari",
    "url": "https://example.com/path-to-font-file.ttf"
    "unicode-range": "U+4??"
  }
]
```

the unicode-range would use the same definition as the CSS specification https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/unicode-range

Reply:
This means we leave the code range setting to the style, hmm, does this necessary? cause only a few languages need shaping the entries won't be that many.